proj/src/main/webapp/chat.js

```diff
@@ -722,23 +722,31 @@ async function addUsernameToMessage(uid, preview) {
 }
 
 // function for when a user adds a picture
-function pfpOnInput() {
+async function pfpOnInput() {
     const input = document.getElementById("pfp-upload");
     const pfp = input.files[0];
 
     const currentUid = firebase.auth().currentUser.uid;
     const pfpStorageRef = firebase.storage().ref(`/profile-pictures/${currentUid}/pfp.png`);
-    pfpStorageRef.put(pfp);
-
-    pfpStorageRef.getDownloadURL().then(function(url) {
+    const uploadStatus = document.getElementById('upload-status')
+    await pfpStorageRef.put(pfp).then(function() {
+        uploadStatus.innerText = 'upload success';
+        setTimeout(function(){ uploadStatus.innerText = ''; }, 5000);
+    }).catch(function(error) {
+        uploadStatus.innerText = 'upload failed: '+error.message;
+        setTimeout(function(){ uploadStatus.innerText = ''; }, 5000);
+    }).then(function() {
+        return pfpStorageRef.getDownloadURL();
+    }).then(function(url) {
         const userPfpRef = firebase.database().ref(`/users/${currentUid}/photo`);
         userPfpRef.set(pfpStorageRef.toString());
 
         userPfp = document.getElementById('user-pfp');
         userPfp.src = url;
     })
     input.files = null; // clear the input
-    sessionStorage[currentUid +" pfp"] = null; // clear session storage
+    input.value = "";
+    delete sessionStorage[currentUid +" pfp"]; // clear session storage
```

Comment:
This change looks like it is from the pfp branch. Is this dependent on those changes?

Reply:
no, i accidentally branched off that branch. i only made small changes so i'll just make another pr